File: AkkaUnitTest/src/main/java/org/akka/essentials/unittest/actors/TickTock.java
package org.akka.essentials.unittest.actors;

import akka.actor.UntypedActor;

public class TickTock extends UntypedActor {

	public static class Tick {
		String message;
		public Tick(String inStr) {
			message = inStr;
		}
	};

	public static class Tock {
		String message;
		public Tock(String inStr) {
			message = inStr;
		}
	};

	public boolean state = false;

	@Override
	public void onReceive(Object message) throws Exception {
		if (message instanceof Tick) {
			tick((Tick) message);
		} else if (message instanceof Tock) {
			tock((Tock) message);
		} else
			throw new IllegalArgumentException("boom!");
	}

	public void tock(Tock message) {
		// do some processing here
		if (state == false)
			state = true;
		else
			state = false;
	}

	public void tick(Tick message) {
		// do some processing here
		sender().tell("processed the tick message");
	}
}

File: AkkaUnitTest/src/main/scala/org/akka/essentials/unittest/example/TickTock.scala
package org.akka.essentials.unittest.example
import akka.actor.Actor

case class Tick(message: String)
case class Tock(message: String)

class TickTock extends Actor {

  var state = false

  def receive: Receive = {
    case message: Tick => tick(message)
    case message: Tock => tock(message)
    case _ => throw new IllegalArgumentException("boom!")
  }

  def tock(message: Tock) = {
    // do some processing here
    if (state == false)
      state = true
    else
      state = false
  }

  def tick(message: Tick) = {
    // do some processing here
    sender.tell("processed the tick message")
  }
}